x/escrow/time.go

```diff
@@ -0,0 +1,21 @@
+package escrow
+
+import (
+	"time"
+
+	"github.com/iov-one/weave"
+)
+
+// isExpired returns true if given time is in the past as compared to the "now"
+// as declared for the block.
+//
+// This function panic if the block time is not provided in the context. This
+// must never happen. The panic is here to prevent from broken setup to be
+// processing data incorrectly.
+func isExpired(ctx weave.Context, t time.Time) bool {
+	blockNow := weave.BlockTime(ctx)
+	if blockNow.IsZero() {
```

Comment:
Ahhh... you do check for zero value here (see comment in the context.go file).
Still, maybe a second return value would make this more explicit:

```go
blockNow, ok := weave.BlockTime(ctx)
if !ok {
  panic("foo")
}
return t.Before(blockNow)
```
```

Reply:
Continued in https://github.com/iov-one/weave/pull/409#discussion_r266382344